### PR TITLE
Break L1s into multiple classes

### PR DESCRIPTION
**Pull request by alex-peck**
* Fix issue preventing L1s from existing in multiple classes
* Break up L1s into multiple classes

### CODE REVIEW

File: src/Test/L1/Worker/VariableL1Tests.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        public async Task SetVariable_ReadVariable()
+        {
+            // Arrange
+            var message = LoadTemplateMessage();
+            // Remove all tasks
+            message.Steps.Clear();
+            // Add variable setting tasks
+            message.Steps.Add(CreateScriptTask("echo \"##vso[task.setvariable variable=testVar]b\""));
+            message.Steps.Add(CreateScriptTask("echo TestVar=$(testVar)"));
+            message.Variables.Add("testVar", "a");
```
Comment (damccorm) [inline src/Test/L1/Worker/VariableL1Tests.cs]: I think you're going to need this change once readonly variables gets merged in - https://github.com/microsoft/azure-pipelines-agent/pull/2771/commits/f259a86fcb6f6e550a9d6667630bb64e0438fd79

File: src/Test/L1/Mock/FakeTaskServer.cs [hunk 2]
```diff
@@ -16,16 +17,24 @@ public Task ConnectAsync(VssConnection jobConnection)
...
-                if (zipName.Equals(taskId.ToString()))
+                if (!Directory.Exists(taskZipsPath))
                 {
-                    return Task.FromResult<Stream>(new FileStream(zip, FileMode.Open));
+                    Directory.CreateDirectory(taskZipsPath);
                 }
+                ZipFile.CreateFromDirectory(taskPath, zip);
+            }
+
+            if (File.Exists(zip))
```
Comment (damccorm) [inline src/Test/L1/Mock/FakeTaskServer.cs]: Maybe add an `else: throw`?

File: src/Misc/externals.sh [hunk 3]
```diff
@@ -187,5 +187,5 @@ fi
 
 if [[ "$L1_MODE" != "" || "$PRECACHE" != "" ]]; then
     # cmdline task
-    acquireExternalTool "$CONTAINER_URL/l1Tasks/d9bafed4-0b18-4f58-968d-86655b4d2ce9.zip" "Tasks/d9bafed4-0b18-4f58-968d-86655b4d2ce9" ""
+    acquireExternalTool "$CONTAINER_URL/l1Tasks/d9bafed4-0b18-4f58-968d-86655b4d2ce9.zip" "Tasks/d9bafed4-0b18-4f58-968d-86655b4d2ce9"
```
Comment (damccorm) [inline src/Misc/externals.sh]: Good catch

File: src/Test/L1/Mock/FakeTaskServer.cs (hunk 2 above)
Comment (alex-peck) [reply src/Test/L1/Mock/FakeTaskServer.cs]: Good idea

File: src/Test/L1/Worker/VariableL1Tests.cs (hunk 1 above)
Comment (alex-peck) [reply src/Test/L1/Worker/VariableL1Tests.cs]: I'll change it here to prevent conflicts

**Comment by damccorm** [comment]
I'm good to merge this when we resolve the conflicts - thanks for doing this

**Comment by stephenmichaelf** [comment]
@alex-peck Where are we on this?

**Comment by alex-peck** [comment]
@stephenmichaelf This magically started working. I think it's ready to merge

**Pull request merged**